conda/plugins/subcommands/doctor/health_checks.py

```diff
@@ -175,6 +181,64 @@ def requests_ca_bundle_check(prefix: str, verbose: bool) -> None:
             )
 
 
+def consistent_env_check(prefix: str, verbose: bool) -> None:
+    # get the plugin manager from context
+    pm = context.plugin_manager
+
+    # get the solver backend from the plugin manager
+    SolverClass = pm.get_solver_backend()
+
+    # get prefix data
+    pd = PrefixData(prefix)
+
+    # create a repodatas dict with noarch subdir already created as it is pre-requisite for creating a channel
+    repodatas = {"noarch": {}}
+
+    # TODO create a list using iter_records and check its length, if 0 exit
+
+    # segregate the package records based on subdir/architecture and package type
+    for record in pd.iter_records():
+        record_data = dict(record.dump())
+        if record.fn.endswith(".conda"):
```

Comment:
let's use `conda.base.constants.CONDA_PACKAGE_EXTENSION_V2`:

```suggestion
        if record.fn.endswith(CONDA_PACKAGE_EXTENSION_V2):
```